dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -367,7 +367,7 @@ await foreach (StreamingChatCompletionsUpdate update in response.ConfigureAwait(
                     }
                 }
 
-                yield return new OpenAIStreamingChatMessageContent(update, update.ChoiceIndex ?? 0, this.DeploymentOrModelName, metadata);
+                yield return new OpenAIStreamingChatMessageContent(update, update.ChoiceIndex ?? 0, this.DeploymentOrModelName, new(metadata));
```

Comment:
Why does this dictionary need to be wrapped?

Reply:
Answered above...

Reply:
Where else is this dictionary instance used other than this one return?

Reply:
This is a yield return, so, changing the dictionary of this chunk will eventually change the same for all other chunks.

If you have a logic like, add key in the `foreach` this would break with key already exists in the following chunk.

Reply:
How about just having:
```C#
metadata ??= new ReadOnlyDictionary<string, object?>(GetResponseMetadata(update));
```
?

Reply:
@stephentoub Currently StreamingContentBase Metadata is a `IDictionary<string, object?>`.

This is read-write to allow changes to the values as contents can be created, serialized and deserialized. 

This is analogous to the `ContentBase` -> `ChatMessageContent` to be able to serialize and deserialize from a ChatHistory.